src/components/pages/Login/Login.js

```diff
@@ -1,77 +1,80 @@
 import React from 'react';
-import { Button } from 'semantic-ui-react';
 import { useState } from 'react';
 import { Redirect } from 'react-router-dom';
 import {
   loginHandler,
-  getToken
+  getUser
 } from '../../../utils/handlers/authenticationHandlers';
 import PropTypes from 'prop-types';
+import LoginForm from './LoginForm';
+import validateLoginForm from './loginValidation';
 
-function Login({ setUser }) {
-  const [credentials, changeCredentials] = useState({
+function Login({ setUser, ...props}) {
+  const [state, setState] = useState({
     username: '',
-    password: ''
+    password: '',
+    errors: {},
+    loginIn: false
   });
 
-  const [status, changeStatus] = useState({
-    loading: false,
-    error: false
-  });
 
-  const submitLogin = async event => {
-    event.preventDefault();
-    changeStatus({ ...status, loading: true });
-    try {
-      const receivedUser = await loginHandler({
-        username: credentials.username,
-        password: credentials.password
-      });
-      changeStatus({ error: false, loading: false });
-      setUser(receivedUser);
-      return <Redirect to="/" />;
-    } catch (err) {
-      changeStatus({ error: true, loading: false });
-    }
+  const handleChange = (e, { name, value }) => {
+    setState(prevState => ({ ...prevState, [name]: value }));
```

Comment:
For a moment there, I thought you were doing `this.setState()`. Luckily, it's all good 😂